docs/docs/best-practices.md

```diff
@@ -55,3 +55,32 @@ enum Modes {
 ```
 
 When using an `Enum`, prefer using a string Enum.
+
+### Producers:
+- only one producer per file
+- they should have a descriptive name (camelCase) and the file should have the same name
+- they should do only one thing
+- they should use `get` instead of `observe` if you don’t need that value to retrigger the producer
+- in the folder structure they should stay close to the components that use them. 
+- if there are more than one producer for a component the propper way of **importing** them should be to by creating an `index.js/ts` file in the producers folder and export them all there and in the component write like this
+``` import * as producers from "./producers";
+...
+<Component>.producers(Object.values(producers)); 
```

Comment:
`Component.producers(producers)`
The fn accepts a producer, an object or an array of producers